### ISSUE TITLE
Sampling interval

### ISSUE THREAD

**Issue by PilotJack** [linkedin/Burrow#394]
Does anyone know how to determine what the sampling interval is for Burrow?  How do we determine this from the configuration file?  How does the windowing work exactly?  What is the time spread between the beginning and end of the window? What is a commit and when does it happen?

Any inputs would be greatly appreciated!

Thanks!

**Comment by rconn01** [comment]
* The sampling interval is defined by the interval you have configured in your [notifiers](https://github.com/linkedin/Burrow/blob/master/config/burrow.toml#L61). 
* The windowing works as described in the [wiki](https://github.com/linkedin/Burrow/wiki/Consumer-Lag-Evaluation-Rules#evaluation-window)
* The time spread is defined differently for each consumer group, it depends how often the consumers in the group are committing. 

Burrow will keep the last X commits from a consumer (defined in [config](https://github.com/linkedin/Burrow/blob/master/config/burrow.toml#L54)). It them applies the [evaluation rules](https://github.com/linkedin/Burrow/wiki/Consumer-Lag-Evaluation-Rules#evaluation-rules). to the window. You can force a minium window size by setting [min.distance](https://github.com/linkedin/Burrow/blob/master/config/burrow.toml#L56) in the event you have consumers that commit at a fast rate

**Comment by jermbop** [comment]
We just upgraded from a very old version of Burrow to the latest commit.  Get Consumer Details is a new endpoint for us.  Going back to PilotJack's question regarding windowing, do you know if the offsets returned by Get Consumer Details are the offsets used by the evaluation rules? 

For example:
I have the following configuration:
[storage.default]
class-name="inmemory"
workers=20
intervals=10
expire-group=604800
min-distance=60

The following response has 10 offsets returned (intervals=10).  Each timestamp is roughly 15 minutes apart. I'm curious as to why the timestamps are 11-15 minutes apart when min-distance=60s.  I would have thought the evaluation window is 10 (intervals) * 60s (min-distance) = 10 minute window, each offset being 60s apart.  Or, is the offset window the interval between offsets used for calculation?

Thanks in advance!

{
  "error": false,
  "message": "consumer detail returned",
  "topics": {
    "my_topic_1": [
      {
        "offsets": [
          {
            "offset": 104675,
            "timestamp": 1526385670719,
            "lag": 0
          },
          {
            "offset": 104681,
            "timestamp": 1526386344643,
            "lag": 99
          },
          {
            "offset": 104682,
            "timestamp": 1526387276249,
            "lag": 114
          },
          {
            "offset": 104683,
            "timestamp": 1526388176786,
            "lag": 136
          },
          {
            "offset": 104684,
            "timestamp": 1526389077684,
            "lag": 159
          },
          {
            "offset": 104685,
            "timestamp": 1526389978170,
            "lag": 182
          },
          {
            "offset": 104687,
            "timestamp": 1526390881456,
            "lag": 259
          },
          {
            "offset": 104688,
            "timestamp": 1526391782650,
            "lag": 272
          },
          {
            "offset": 104689,
            "timestamp": 1526392683073,
            "lag": 286
          },
          {
            "offset": 104690,
            "timestamp": 1526393215361,
            "lag": 326
          }
        ],
        "owner": "/my-owner-ip",
        "current-lag": 425
      }
}

**Comment by rconn01** [comment]
they are the offsets used for the calculation. `min-distance` just defines the minimum distance commits can be. The window is ultimately defined by how frequently the consumer in the group is committing its offsets. If the consumer is only committing every 15 minutes, that will be the delta between the commits in the window. If the consumer is committing faster than the min-distance, burrow will only accept a commit once per min-distance.

**Comment by PilotJack** [comment]
rconn01 -- I don't understand your answer.  And windowing is not really described in detail the wiki.  Here's how I'd answer the question if I were in your shoes:

A burrow commit is defined as <fill in the blank>.  Here's what happens during a commit <fill in the blank> Burrow samples bla bla bla times per minute depending on the bla bla parameter>.  You configure commit interval using the following parameters <param1 for this, param 2 for that>.  Here's an example - suppose we set up the following configuration parameters <param1=this, param2=that, param3=the other thing>.  With those parameters, this is exactly what Burrow would do:  It would sample over a window of <whatever> seconds. Over that window period, it would sample the queues <whatever> times.  This means you can assume a sampling interval of <a number> seconds.
